samples/dnn/optical_flow.py

```diff
@@ -17,17 +23,27 @@
 
 
 class OpticalFlow(object):
-    def __init__(self, proto, model, height, width):
```

Comment:
Use readNet for every model. Proto might be empty string by default (choose other default argument values too)

Reply:
I tried to use `readNet` for both *FlowNet* and *RAFT* but the `readNet` unlike `readNetFromCaffe` takes the config/prototxt file not as string but as a path. A work around this which I thought can solve this is to create a new copy of prototxt because we cannot replace the values in the original file because because we won't be able to use it for any other size. Please take a look at this code (if this is good) and do tell me if there's any other way to approach this problem.

In my opinion this will slow down the code, but I can't seem to find any other way around without writing over the placeholders in the original prototxt file 

```python
config = ''
proto = open(args.proto).readlines()
for line in proto:
    for key, value in var.items():
        tag = "$%s$" % key
        line = line.replace(tag, str(value))
    config += line

with open('config_run.prototxt', 'w') as f:
    f.writelines(config)

opt_flow = OpticalFlow(args.model, var['ADAPTED_HEIGHT'], var['ADAPTED_WIDTH'], 'config_run.prototxt')
```

Reply:
`readNet` can accept a path, there is no problem with it. Please try again

Reply:
Yes, I know readNet takes a path, but the previous example was using readNetFromCaffe in which it was passing a string not a path. And when passing `readNet` the path it's throwing an error, because the prototxt has parameters that are supposed to passed. 
> `readNet` unlike `readNetFromCaffe` takes the config/prototxt file not as string but as a path

Now, to solve this problem without hard coding the parameters in the prototxt file we are creating another file of which we'll pass the path. 

And I didn't say `readNet` can't take path, I just said it needs parameters first and to not hard coding in the original config file we can make another

Reply:
Ah, I see. Do you mean that `readNet` cannot accept path+bin array while `readNetFromCaffe` can? If so, I recommend to do the following in constructor:
```
if proto:
    self.net = cv.dnn.readNetFromCaffe(proto, model)
else:
    self.net = cv.dnn.readNet(model)
```
It will help avoid dumping prototxt content to `config_run.prototxt`

Reply:
> Do you mean that readNet cannot accept path+bin array while readNetFromCaffe can?
Yes, that's why I used `readNetFromCaffe` in the first commit. Okay, I'll do the needful.